pkg/cmd/release/update_versions.go

```diff
@@ -282,27 +285,20 @@ func updateVersions(_ *cobra.Command, _ []string) error {
 	// This way we can fail early and avoid unnecessary work closing the PRs we were able to create.
 	log.Printf("repos to work on: %s\n", reposToWorkOn)
 	var prs []string
+	var workOnRepoErrors []error
 	for _, repo := range reposToWorkOn {
-		log.Printf("Cloning repo %s", repo.name())
-		if err := repo.clone(); err != nil {
-			return fmt.Errorf("cannot clone %s: %w", repo.name(), err)
-		}
-		log.Printf("Branching repo %s", repo.name())
-		if err := repo.checkout(); err != nil {
-			return fmt.Errorf("cannot create branch %s: %w", repo.name(), err)
-		}
-		log.Printf("Munging repo %s", repo.name())
-		if err := repo.apply(); err != nil {
-			return fmt.Errorf("cannot mutate repo %s: %w", repo.name(), err)
-		}
-		log.Printf("commiting changes to repo %s", repo.name())
-		if err := repo.commit(); err != nil {
-			return fmt.Errorf("cannot commit changes in repo %s: %w", repo.name(), err)
+		repo.workOnRepoError = workOnRepo(repo)
+		if repo.workOnRepoError != nil {
+			log.Printf("workOnRepo: error occurred while working on %s: %s", repo.name(), repo.workOnRepoError)
+			workOnRepoErrors = append(workOnRepoErrors, repo.workOnRepoError)
 		}
 	}
 
 	// Now that our local changes are staged, we can try and publish them.
 	for _, repo := range reposToWorkOn {
+		if repo.workOnRepoError != nil {
+			continue
+		}
 		dest := path.Join(globalWorkDir, repo.checkoutDir())
 		// We avoid creating duplicated PRs to allow this command to be
 		// run multiple times.
```

Comment:
> If this needs to be rerun, will it be safe for all the repos what had previously run successfully for the given version?

I believe so, based on the comment on line 307 (was 303):

		// We avoid creating duplicated PRs to allow this command to be
		// run multiple times.